DXMainClient/DXGUI/Multiplayer/CnCNet/GameCreationWindow.cs

```diff
@@ -239,17 +240,14 @@ private void BtnLoadMPGame_LeftClick(object sender, EventArgs e)
 
         private void BtnCreateGame_LeftClick(object sender, EventArgs e)
         {
-            string gameName = tbGameName.Text.Replace(";", string.Empty);
 
-            if (string.IsNullOrEmpty(gameName))
-            {
-                return;
-            }
+            var lobbyName = tbGameName.Text;
+            var lobbyNameValid = NameValidator.IsLobbyNameValid(lobbyName);
 
-            if (new ProfanityFilter().IsOffensive(gameName))
+            if (!string.IsNullOrEmpty(lobbyNameValid))
             {
-                XNAMessageBox.Show(WindowManager, "Offensive game name".L10N("Client:Main:GameNameOffensiveTitle"),
-                    "Please enter a less offensive game name.".L10N("Client:Main:GameNameOffensiveText"));
+                XNAMessageBox.Show(WindowManager, "Invalid lobby name".L10N("Client:Main:GameNameInvalid"),
+                    lobbyNameValid);
```

Comment:
Hmm, the name collision between `Invalid lobby name` and `Client:Main:GameNameInvalid` looks odd to me. Maybe we should use `game*name` instead of `lobby*name` when referring to the game house?